iox_catalog/src/interface.rs

```diff
@@ -1708,78 +1722,155 @@ pub(crate) mod test_helpers {
         assert_eq!(skipped_compactions[0].reason, "I'm on fire");
         assert_eq!(skipped_compactions[0].num_files, 11);
         assert_eq!(skipped_compactions[0].limit_num_files, 12);
         assert_eq!(skipped_compactions[0].estimated_bytes, 110);
         assert_eq!(skipped_compactions[0].limit_bytes, 120);
         //
-        let skipped_partition_record = repos
+        let skipped_partition_records = repos
             .partitions()
-            .get_in_skipped_compaction(other_partition.id)
+            .get_in_skipped_compactions(&[to_skip_partition.id])
             .await
-            .unwrap()
             .unwrap();
-        assert_eq!(skipped_partition_record.partition_id, other_partition.id);
-        assert_eq!(skipped_partition_record.reason, "I'm on fire");
+        assert_eq!(
+            skipped_partition_records[0].partition_id,
+            to_skip_partition.id
+        );
+        assert_eq!(skipped_partition_records[0].reason, "I'm on fire");
 
-        // Delete the skipped compaction
+        // Can receive multiple skipped compactions for different partitions
+        repos
+            .partitions()
+            .record_skipped_compaction(
+                to_skip_partition_too.id,
+                "I am le tired too",
+                1,
+                2,
+                4,
+                10,
+                20,
+            )
+            .await
+            .unwrap();
+        let skipped_compactions = repos.partitions().list_skipped_compactions().await.unwrap();
+        assert_eq!(skipped_compactions.len(), 2);
+        assert_eq!(skipped_compactions[0].partition_id, to_skip_partition.id);
+        assert_eq!(
+            skipped_compactions[1].partition_id,
+            to_skip_partition_too.id
+        );
+        // confirm can fetch subset of skipped compactions (a.k.a. have two, only fetch 1)
+        let skipped_partition_records = repos
+            .partitions()
+            .get_in_skipped_compactions(&[to_skip_partition.id])
+            .await
+            .unwrap();
+        assert_eq!(skipped_partition_records.len(), 1);
+        assert_eq!(skipped_compactions[0].partition_id, to_skip_partition.id);
+        let skipped_partition_records = repos
+            .partitions()
+            .get_in_skipped_compactions(&[to_skip_partition_too.id])
+            .await
+            .unwrap();
+        assert_eq!(skipped_partition_records.len(), 1);
+        assert_eq!(
+            skipped_partition_records[0].partition_id,
+            to_skip_partition_too.id
+        );
+        // confirm can fetch both skipped compactions, and not the unskipped one
+        let skipped_partition_records = repos
+            .partitions()
+            .get_in_skipped_compactions(&[
+                partition.id,
+                to_skip_partition.id,
+                to_skip_partition_too.id,
+            ])
+            .await
+            .unwrap();
```

Comment:
I'm glad this test is useful! Can we add one more value here that should *not* be returned (and also not cause any errors), the [`non_existing_partition_id`](https://github.com/influxdata/influxdb_iox/blob/1138a5907c449b595b35d3fb6da7b078de552b97/iox_catalog/src/interface.rs#L1523)? So change this call to:

```suggestion
        // confirm can fetch both skipped compactions, and not the unskipped one
        let skipped_partition_records = repos
            .partitions()
            .get_in_skipped_compactions(&[
                partition.id,
                to_skip_partition.id,
                to_skip_partition_too.id,
                non_existing_partition_id,
            ])
            .await
            .unwrap();
```

but leave the assertions as-is.